.github/workflows/ci.yml

```diff
@@ -0,0 +1,104 @@
+name: CI
+
+env:
+  JAVA8: adopt@1.8
+  JAVA11: adopt@1.11
+  SCALA_211: 2.11.12
+  SCALA_212: 2.12.13
+  SCALA_213: 2.13.5
+  SCALA_3: 3.0.0-RC1
+  JAVA_OPTS: -Xms2048M -Xmx3072M -Xss6M -XX:ReservedCodeCacheSize=256M -Dfile.encoding=UTF-8
+  JVM_OPTS:  -Xms2048M -Xmx3072M -Xss6M -XX:ReservedCodeCacheSize=256M -Dfile.encoding=UTF-8
+on:
+  pull_request:
+  push:
+    branches: ['master']
+  release:
+    types:
+      - published
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        java: ['adopt@1.8', 'adopt@1.11']
+        scala: ['2.12.13', '2.13.5']
+    steps:
+    - uses: actions/checkout@v2.3.4
+    - uses: olafurpg/setup-scala@v10
+      with:
+        java-version: ${{ matrix.java }}
+    - name: Cache scala dependencies
+      uses: coursier/cache-action@v6
+    - name: Lint code
+      run: sbt fmtCheck fixCheck
+    - name: Mima Checks
+      run: sbt ++${{ matrix.scala }}! mimaChecks
+    - name: Run JVM tests
+      run: sbt ++${{ matrix.scala }}! testJVM
+    - name: Run JS Tests
+      run: sbt ++${{ matrix.scala }}! testJS
+    - name: Run Scala Native Tests
+      run: sbt testNative
```

Comment:
Maybe we should extract these jobs? I think we have the similar thing in prelude, but I'm not sure. It would minimize extra steps (e. g. native builds and linting run in matrix).

Reply:
sure I am good with that ... should we merge and do that as subsequent PR or shall we make the changes in this PR?

Reply:
Let's merge and decommission CircleCI, i can follow up on it today.

Reply:
Awesome plan thank you sir 🍷